### DISCUSSION TITLE
How to access widget using their `id` property?

### DISCUSSION THREAD

**Discussion by dr-sandipan-chowdhury** [beeware/toga#2134] [Debugging help]
Was not able to find a way to use the `id` property to access the widget. Each time when I do need access to widget, I must save the widget in some kind of variable. I want some feature like "document.getElementById" of Javascript. 
I dont know if it is implemented or not...

**Comment by dr-sandipan-chowdhury** [comment]
Is this feature implemented in toga yet?

**Comment by freakboy3742** [comment]
`window.widgets[some_id]` will return a widget that is currently part of a given window.

`app.widgets[some_id]` does the same lookup, over all windows of the app.

However, to be clear: "saving the widget in some kind of variable" is... programming. While ID-based lookup can be convenient in some situations, and is *necessary* for complex meta-programming cases, for simple apps, storing the widgets you're using as variables is good programming practice, not something to be avoided.

**Reply by dr-sandipan-chowdhury** [reply]
Thanks Russell.
